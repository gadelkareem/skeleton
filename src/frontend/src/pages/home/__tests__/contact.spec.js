import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import initData from '@@/globals/init-data'
import Vuetify from 'vuetify'
import Page from '../contact'

const localVue = createLocalVue()

describe('invoices.vue', () => {
  let store, vuetify

  beforeEach(() => {
    const actions = {
      'page/title': jest.fn(),
      'loading/start': jest.fn(),
      'loading/finish': jest.fn()
    }
    const getters = {
      'user/user': () => initData().user
    }
    store = new Vuex.Store({
      actions,
      getters,
      state: {
        loading: { status: false }
      }
    })
    vuetify = new Vuetify()
  })

  it('should match snapshot', () => {
    const router = new VueRouter({})
    const w = mount(Page, {
      store,
      router,
      localVue,
      vuetify,
      stubs: ['router-link', 'router-view'],
      created () {
        this.$vuetify.lang = {
          t: () => {}
        }
        this.$vuetify.breakpoint = {
          width: 0,
          height: 0
        }
        this.$vuetify.theme = { dark: false }
      }
    })

    expect(w.html()).toMatchSnapshot()
    w.destroy()
  })
})
