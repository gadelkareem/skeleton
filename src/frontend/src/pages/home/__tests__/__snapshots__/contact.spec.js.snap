// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invoices.vue should match snapshot 1`] = `
<section class="container container--fluid">
  <div class="row justify-center">
    <div class="v-card--material pa-3 pa-6 v-card v-sheet theme--light v-card--material--has-heading" style="min-width: 70%;" min-width="70%">
      <div class="d-flex grow flex-wrap">
        <div class="text-start v-card--material__heading mb-n6 v-sheet theme--dark elevation-6 info pa-7" style="width: 100%;">
          <div class="display-2 font-weight-light">
            Contact
          </div>
          <div class="subtitle-1 font-weight-light">
            Fill in your message
          </div>
          <!---->
          <!---->
        </div>
        <!---->
      </div> <span> <!----> <!----></span>
      <form novalidate="novalidate" method="post" class="v-form pt-8">
        <div class="container py-0">
          <div class="row">
            <div class="v-input theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span>​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-9" class="v-label theme--light" style="left: 0px; position: absolute;">Name*</label><input data-username="" id="input-9" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="row">
            <div class="v-input theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span>​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-12" class="v-label theme--light" style="left: 0px; position: absolute;">Email*</label><input id="input-12" type="email"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="row">
            <div class="v-input v-textarea theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span>​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-15" class="v-label theme--light" style="left: 0px; position: absolute;">Message*</label><textarea id="input-15" rows="5"></textarea></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="spacer"></div>
          <div class="text-right col"><button type="submit" class="mr-0 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default info"><span class="v-btn__content">
                Send
              </span></button></div>
        </div>
      </form>
      <!---->
    </div>
  </div>
</section>
`;
